examples/SquareLine/WiFiClock/WiFiClock.ino

```diff
@@ -0,0 +1,980 @@
+        lv_textarea_set_text(ui_TextPassword, "");
+
+        lvgl_port_unlock();
+    }
+}
+
+void handleAlarmActivation()
+{
+    static const char * btns[] ={""};
+
+    lvgl_port_lock(-1);
+
+    lv_obj_t* Alarm_mbox = lv_msgbox_create(NULL, "Alarm activated", "Click the button to turn off the alarm", btns, true);
+    lv_obj_center(Alarm_mbox);
+
+    alarm_flag = false;
+    Serial.println("alarm_flag: false");
+
+    lvgl_port_unlock();
+}
+
+void scanAndDisplayWiFiList()
+{
+    if(wifi_list_switch) {
+        num_wifi = WiFi.scanNetworks();
+        Serial.println("Scan done");
+
+        if(num_wifi == 0) {
+            Serial.println("no networks found");
+        } else if(wifi_list_switch) {
+            wifi_list_switch = false;
+            Serial.println("wifi_list_switch: false");
+
+            displayWifiList();
+        }
+        WiFi.scanDelete();
+    }
+}
+
+void handleWifiConnection()
+{
+    if(password_valid_flag) {
+        Serial.println("Wifi connecting...");
+        Serial.printf("%s\n",  selected_wifi_name);
+        Serial.printf("%s\n", wifi_password);
+
+        handlePasswordValidation();
+    }
+}
+
+void updateWeatherAndTime()
+{
+    if(wifi_connected_flag) {
+        timeClient.update();
+        get_weather_count++;
+        unsigned long epochTime = timeClient.getEpochTime();
+        struct tm *ptm = gmtime((time_t *)&epochTime);
+        snprintf(date_string, sizeof(date_string), "%04d-%02d-%02d", ptm->tm_year + 1900, ptm->tm_mon + 1, ptm->tm_mday);
+
+        if(get_weather_count >= 120) {
+            ParseWeather(weather_url);
+            get_weather_count = 0;
+        }
+
+        struct timeval tv = {
+            .tv_sec = epochTime - 28800,
+            .tv_usec = 0,
+        };
+        settimeofday(&tv, NULL);
+
+        if(WiFi.status() != WL_CONNECTED) {
+            wifi_connected_flag = false;
+            Serial.println("wifi_connected_flag: false");
+            writeIntToNVS("NVS_WifiFg", wifi_connected_flag);
+            http.end();
+            WiFi.disconnect();
+        }
+
+        if(lv_obj_has_flag(ui_ImageWifi, LV_OBJ_FLAG_HIDDEN)) {
+            lvgl_port_lock(-1);
+            _ui_flag_modify(ui_ImageWifi, LV_OBJ_FLAG_HIDDEN, _UI_MODIFY_FLAG_REMOVE);
+            lvgl_port_unlock();
+        }
+    } else {
+        time(&now);
+        localtime_r(&now, &timeinfo);
+        strftime(time_str, sizeof(time_str), "%H:%M:%S", &timeinfo);
+        strftime(date_string, sizeof(date_string), "%Y-%m-%d", &timeinfo);
+        strftime(weekday_str, sizeof(weekday_str), "%A", &timeinfo);
+
+        if(!lv_obj_has_flag(ui_ImageWifi, LV_OBJ_FLAG_HIDDEN)) {
+            lvgl_port_lock(-1);
+            _ui_flag_modify(ui_ImageWifi, LV_OBJ_FLAG_HIDDEN, _UI_MODIFY_FLAG_ADD);
+            lvgl_port_unlock();
+        }
+    }
+}
+
+void triggerAlarm()
+{
+    if(alarm_flag) {
+        handleAlarmActivation();
+    }
+}
```

Comment:
I think this function is too short and should just merge with `handleAlarmActivation()`.